modules/light-clients/08-wasm/types/querier_test.go

```diff
@@ -157,6 +161,81 @@ func (suite *TypesTestSuite) TestStargateQuery() {
 				})
 			},
 		},
+		{
+			"success: verify membership query",
```

Comment:
This test sets a key-value in the ibc store, using `"mock-key": "mock-value"`.  

It registers the query callback for `VerifyMembership` on the `Status` call to the wasm client using the mockVM. 
It also needs to register a callback for `VerifyMembership` since this is all mocked, we can still unmarshal and verify it tho within the callback registered for `VerifyMembership`.

The only alternative I see to this is setting up an additional ibctesting path with a regular tendermint client, and querying some proof to be verified by that using its client ID as the target. Given that we use a custom `wasmtesting.Endpoint` for these tests, I decided not to mix and match endpoint functionality between testing pkgs. I think the tests serves it purpose, it routes, decodes and verifies a proof.

Reply:
Maybe for clarification you could add a comment of why the `VerifyMembership` callback is needed in the test and explain how the flow goes?

Reply:
I added a comment :)